src/js/components/timeslot.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import {
+  DEFAULT,
+  SELECTED,
+  DISABLED,
+ } from 'constants/timeslot.js';
+
+export default class Timeslot extends React.Component {
+  render() {
+    const {
+      description,
+      status,
+      onClick,
+    } = this.props;
+
+    const timeslotClassNames = classnames({
+      'tsc-timeslot': true,
+      'tsc-timeslot--selected': status == SELECTED,
+      'tsc-timeslot--disabled': status == DISABLED,
+    });
+
+    return (
+      <div className= { timeslotClassNames } onClick = {onClick}>
+        {description}
+      </div>
+    );
+  }
+}
+
```

Comment:
add `defaultProps` to `status` as `DEFAULT`. It doesn't need to be required though - the app won't fail to work if it is `null` or `undefined` either way.